src/unix/linux-core.c

```diff
@@ -552,15 +552,51 @@ int uv_uptime(double* uptime) {
 }
 
 
+static int uv__cpu_num(int* statfile_fd, unsigned int* numcpus) {
+  unsigned int num;
+  char buf[1024];
+  FILE* fp;
+
+  lseek(*statfile_fd, 0, SEEK_SET);
+  fp = fdopen(*statfile_fd, "r");
```

Comment:
You can avbvoid having to `fdopen` twice by passing around the `FILE*` stream and using `rewind` or `fseek`.